functions/security-definitions.js

```diff
@@ -0,0 +1,67 @@
+// Check API definition to ensure conformance to Azure security schemes guidelines.
+
+// Check:
+// - There is at least one security scheme.
+// - All security schemes are either:
+//   - type: oauth2 or
+//   - type: apiKey with in: header
+// - An oauth2 security scheme defines at least one scope.
+// - All scopes defined in an oauth2 security scheme match a pattern:
+//   - https:\/\/[\w-]+(\.[\w-]+)+/[\w-.]+
+
+// @param doc - the entire API document
+module.exports = (doc) => {
+  if (doc === null || typeof doc !== 'object') {
+    return [];
+  }
+
+  if (!doc.securityDefinitions || (typeof doc.securityDefinitions === 'object' && Object.keys(doc.securityDefinitions).length === 0)) {
+    return [{
+      message: 'At least one security scheme must be defined.',
+      path: ['securityDefinitions'],
+    }];
+  }
+
+  const schemes = doc.securityDefinitions;
+
+  const errors = [];
+
+  Object.keys(schemes).forEach((schemeKey) => {
+    const scheme = schemes[schemeKey];
+    if (typeof scheme === 'object') {
```

Comment:
I assume there is some validator for the schema overall at a higher level? In case if scheme is not an object for example.

Reply:
Yes, but I'll add a comment here to avoid confusion.